hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -944,13 +944,15 @@ public void startCommitWithTime(String instantTime, String actionType) {
    * Starts a new commit time for a write operation (insert/update/delete) with specified action.
    */
   private void startCommitWithTime(String instantTime, String actionType, HoodieTableMetaClient metaClient) {
+    HoodieTableMetaClient updatedMetaClient = metaClient;
     if (needsUpgradeOrDowngrade(metaClient)) {
       // unclear what instant to use, since upgrade does have a given instant.
       executeUsingTxnManager(Option.empty(), () -> tryUpgrade(metaClient, Option.empty()));
+      updatedMetaClient = createMetaClient(true);
```

Comment:
We better update the options within `tryUpgrade`, like what we already do: `reloadTableConfig`, `reloadActiveTimeline`.

Reply:
The metaClient is locally passed to this method which in turn needs to be passed to startCommit. On upgrade, metaclient is stale (wrong layoutVetsion) and needs to be updated. This is the reason for the above change.

Reply:
Can we just refresh the `timelineLayoutVersion` and `timelineLayout` from the table config right after the table config and timeline are reloaded?